### PR TITLE
feat(DIPK): make the gene expression autoencoder configurable

### PR DESCRIPTION
**Pull request by jfrog64**
# PR Checklist for all PRs

- [x] This comment contains a description of changes (with reason)
- [ ] Referenced issue is linked — no existing issue, this came out of trying to warm start the DIPK autoencoder from an external cohort
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated — not needed, `drevalpy.models.DIPK.gene_expression_encoder` is already an `automodule`, so the new parameters and their docstrings show up by themselves; `_fit_gene_encoder` is private

### New features

`train_gene_expession_autoencoder` hard-codes the whole training setup of the DIPK gene expression
autoencoder in its body:

```python
lr = 1e-4
batch_size = 1024
...
patience = 3
```

Everything here is additive and the defaults are exactly those values, so **DIPK trains bit-for-bit as
before** when nothing is passed. What becomes possible:

* `lr`, `batch_size`, `patience` are now keyword parameters. The `batch_size` docstring spells out the pitfall
  that cost us a while: the DIPK default of 1024 is chosen for the *response row* matrix, in which each cell
  line is repeated once per response. A caller that trains the autoencoder on the unique cell lines instead
  — a few hundred rows — gets a single gradient step per epoch with that default and has to lower it.
* `encoder_state` / `decoder_state` accept state dicts, which turns the call into fine tuning of pretrained
  weights instead of training from scratch. This is what we needed: pretrain the autoencoder on a large
  external expression cohort, then fine tune it on the cell lines of the current fold.
* `DIPKModel.train` now obtains the encoder through a small `_fit_gene_encoder` hook instead of calling the
  training function inline. A subclass can override that one method to warm start, freeze, or replace the
  encoder without copying the whole 60-line `train` method. The hook itself only forwards
  `epochs_autoencoder`, so the default path is unchanged.

Tests: `tests/test_dipk_autoencoder_parameters.py`

* the defaults of the new parameters are the previously hard-coded values, so nothing changes silently
* warm starting reproduces the given weights in the returned encoder, for the encoder alone and for encoder
  plus decoder
* `lr` and `batch_size` reach the optimizer and the `DataLoader`
* `patience` decides when early stopping triggers, checked against the number of epochs actually run with a
  constant-loss encoder/decoder pair so that improvement never happens
* `DIPKModel.train` goes through `_fit_gene_encoder` and forwards the `epochs_autoencoder` hyperparameter

### Maintenance

Note on overlap: #466 changes the memory handling inside the same function (`train_gene_expession_autoencoder`).
The two changes are independent in intent but touch neighbouring lines, so whichever is merged second needs a
small rebase. Happy to do that on this branch as soon as the order is clear.

### CODE REVIEW

File: drevalpy/models/DIPK/dipk.py [hunk 1]
```diff
@@ -79,6 +79,25 @@ def build_model(self, hyperparameters: dict[str, Any]) -> None:
...
+        from weights pretrained on an external cohort) without duplicating the whole train() method.
+
+        :param train_gene_expression: gene expression of the training rows
+        :param val_gene_expression: gene expression of the early stopping rows
+        :returns: trained gene expression encoder
+        """
+        return train_gene_expession_autoencoder(
+            train_gene_expression,
+            val_gene_expression,
+            epochs_autoencoder=self.hyperparameters["epochs_autoencoder"],
```
Comment (JudithBernett) [inline drevalpy/models/DIPK/dipk.py:118]: Wouldn't the hyperparameters have to be forwarded here? otherwise lr, patience and batch size are not passed to the model, right?

**Review by JudithBernett — CHANGES_REQUESTED** [review]

File: drevalpy/models/DIPK/dipk.py (hunk 1 above)
Comment (jfrog64) [reply drevalpy/models/DIPK/dipk.py:118]: You're right, thanks — as it stood, `lr`, `patience` and `batch_size` were only reachable by subclassing, not from the hyperparameters. Fixed in bba0ca3.

There was one snag: DIPK's `hyperparameters.yaml` already uses all three names for the prediction network, with different values — `batch_size: 64` vs. the autoencoder's 1024 and `patience: 10` vs. 3 (`lr` happens to be 0.0001 in both). Forwarding `hyperparameters["batch_size"]` here would therefore have silently retrained the autoencoder with batch size 64 and patience 10, i.e. changed existing results.

So the keys carry the `_autoencoder` suffix of the `epochs_autoencoder` key that was already there:

```yaml
lr_autoencoder: [0.0001]
patience_autoencoder: [3]
batch_size_autoencoder: [1024]
```

and the hook reads them with `self.hyperparameters.get("lr_autoencoder", 1e-4)` etc., so a hyperparameter set without them keeps the previous hard-coded values and nothing changes numerically. All three values are the current ones, and each list has a single entry, so the tuning grid doesn't grow either.

Tests: the hook test now checks all four parameters, plus one for the fallback and one asserting that the predictor's `lr`/`patience`/`batch_size` do *not* leak into the autoencoder. Happy to flip the names round to `autoencoder_lr` etc. if you prefer that ordering.

**Comment by jfrog64** [comment]
Also merged current `development` into the branch — #466 landed in the meantime and touched the same autoencoder function, so the PR had gone conflicting. The conflict itself was harmless: the host-memory change and the new parameters combine directly, and `_validation_loss` now uses the configurable `batch_size` for its chunking. The only manual resolution was the `build_model` docstring, where the `gene_list` entry from #469 and the autoencoder entries both belong.

Locally the suite is at 253 passed, with the two pre-existing `test_dataset.py` failures that need the `CurveCurator` binary, which isn't in my environment.

**Review by JudithBernett — APPROVED** [review]
